From 76f089563f67f251901c4d0d196b40cf5f3840a3 Mon Sep 17 00:00:00 2001
From: Fabio Bertinatto <fbertina@redhat.com>
Date: Mon, 3 Apr 2023 14:44:02 -0300
Subject: [PATCH] UPSTREAM: <carry>: APISelfSubjectReview: only test v1beta1
 API

---
 test/e2e/auth/selfsubjectreviews.go | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/test/e2e/auth/selfsubjectreviews.go b/test/e2e/auth/selfsubjectreviews.go
index c617edd9806..e309c3d78e4 100644
--- a/test/e2e/auth/selfsubjectreviews.go
+++ b/test/e2e/auth/selfsubjectreviews.go
@@ -42,10 +42,8 @@ var _ = SIGDescribe("SelfSubjectReview", func() {
 			Testname: SelfSubjectReview API
 			Description:
 			The authentication.k8s.io API group MUST exist in the /apis discovery document.
-			The authentication.k8s.io/v1alpha1 API group/version MUST exist in the /apis/mode.k8s.io discovery document.
 		    The authentication.k8s.io/v1beta1 API group/version MUST exist in the /apis/mode.k8s.io discovery document.
 			The authentication.k8s.io/v1 API group/version MUST exist in the /apis/mode.k8s.io discovery document.
-			The selfsubjectreviews resource MUST exist in the /apis/authentication.k8s.io/v1alpha1 discovery document.
 			The selfsubjectreviews resource MUST exist in the /apis/authentication.k8s.io/v1beta1 discovery document.
 			The selfsubjectreviews resource MUST exist in the /apis/authentication.k8s.io/v1 discovery document.
 			The selfsubjectreviews resource MUST support create.
@@ -107,7 +105,8 @@ var _ = SIGDescribe("SelfSubjectReview", func() {
 				}
 			}
 		},
-		ginkgo.Entry("authentication/v1alpha1", "v1alpha1", authenticationv1alpha1.SchemeGroupVersion.String()),
+		// OpenShift: Skip v1alpha check
+		// ginkgo.Entry("authentication/v1alpha1", "v1alpha1", authenticationv1alpha1.SchemeGroupVersion.String()),
 		ginkgo.Entry("authentication/v1beta1", "v1beta1", authenticationv1beta1.SchemeGroupVersion.String()),
 		ginkgo.Entry("authentication/v1", "v1", authenticationv1.SchemeGroupVersion.String()),
 	)
@@ -116,6 +115,9 @@ var _ = SIGDescribe("SelfSubjectReview", func() {
 		// Check creating
 		ginkgo.By("creating SSR authentication/v1alpha1")
 		{
+			// OpenShift: Skip v1alpha check
+			ginkgo.Skip("No authentication/v1alpha1 available")
+
 			// Use impersonate to make user attributes predictable
 			config := restConfig(f)
 
-- 
2.43.0

