From ac28fae9d208183f8a27410475fe5a7dd2cc48ec Mon Sep 17 00:00:00 2001
From: Fabio Bertinatto <fbertina@redhat.com>
Date: Mon, 12 Jun 2023 15:34:19 -0300
Subject: [PATCH] UPSTREAM: <drop>: disable transient dep check

---
 hack/update-vendor.sh | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/hack/update-vendor.sh b/hack/update-vendor.sh
index abc672df80a..61b50a75a14 100755
--- a/hack/update-vendor.sh
+++ b/hack/update-vendor.sh
@@ -353,14 +353,14 @@ comm -23 \
 while read -r X; do echo "-dropreplace=${X}"; done |
 xargs -L 100 go mod edit -fmt
 
-# disallow transitive dependencies on k8s.io/kubernetes
-loopback_deps=()
-kube::util::read-array loopback_deps < <(go mod graph | grep ' k8s.io/kubernetes' || true)
-if [[ -n ${loopback_deps[*]:+"${loopback_deps[*]}"} ]]; then
-  kube::log::error "Disallowed transitive k8s.io/kubernetes dependencies exist via the following imports:" >&22 2>&1
-  kube::log::error "${loopback_deps[@]}" >&22 2>&1
-  exit 1
-fi
+# # disallow transitive dependencies on k8s.io/kubernetes
+# loopback_deps=()
+# kube::util::read-array loopback_deps < <(go mod graph | grep ' k8s.io/kubernetes' || true)
+# if [[ -n ${loopback_deps[*]:+"${loopback_deps[*]}"} ]]; then
+#   kube::log::error "Disallowed transitive k8s.io/kubernetes dependencies exist via the following imports:" >&22 2>&1
+#   kube::log::error "${loopback_deps[@]}" >&22 2>&1
+#   exit 1
+# fi
 
 # Phase 6: add generated comments to go.mod files
 kube::log::status "go.mod: adding generated comments" >&11
-- 
2.40.1

