From c19af2b651e06ccede715c5f0bb2fa4746827fb6 Mon Sep 17 00:00:00 2001
From: Fabio Bertinatto <fbertina@redhat.com>
Date: Wed, 28 Jun 2023 17:40:29 -0300
Subject: [PATCH] UPSTREAM: <drop>: update feature gates

---
 pkg/features/kube_features.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pkg/features/kube_features.go b/pkg/features/kube_features.go
index 124b85972cf..474aee0e9b6 100644
--- a/pkg/features/kube_features.go
+++ b/pkg/features/kube_features.go
@@ -1018,7 +1018,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	RecoverVolumeExpansionFailure: {Default: false, PreRelease: featuregate.Alpha},
 
-	RetroactiveDefaultStorageClass: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.29
+	RetroactiveDefaultStorageClass: {Default: true, PreRelease: featuregate.GA, LockToDefault: false}, // remove in 1.29
 
 	RotateKubeletServerCertificate: {Default: true, PreRelease: featuregate.Beta},
 
-- 
2.41.0

