From 5825653068cfd116f9b41403ea4b910cded1b553 Mon Sep 17 00:00:00 2001
From: Peter Hunt <pehunt@redhat.com>
Date: Mon, 25 Nov 2024 16:58:59 -0500
Subject: [PATCH] UPSTREAM: <carry>: features: add MinimumKubeletVersion
 feature

Signed-off-by: Peter Hunt <pehunt@redhat.com>
---
 pkg/features/openshift_features.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/pkg/features/openshift_features.go b/pkg/features/openshift_features.go
index 6325606ee88..3ba55fdac90 100644
--- a/pkg/features/openshift_features.go
+++ b/pkg/features/openshift_features.go
@@ -5,6 +5,7 @@ import (
 )
 
 var RouteExternalCertificate featuregate.Feature = "RouteExternalCertificate"
+var MinimumKubeletVersion featuregate.Feature = "MinimumKubeletVersion"
 
 // registerOpenshiftFeatures injects openshift-specific feature gates
 func registerOpenshiftFeatures() {
@@ -12,4 +13,8 @@ func registerOpenshiftFeatures() {
 		Default:    false,
 		PreRelease: featuregate.Alpha,
 	}
+	defaultKubernetesFeatureGates[MinimumKubeletVersion] = featuregate.FeatureSpec{
+		Default:    false,
+		PreRelease: featuregate.Alpha,
+	}
 }
-- 
2.49.0

