From 97a47bc96679f082d76efa58982c93c4688cc803 Mon Sep 17 00:00:00 2001
From: Fabio Bertinatto <fbertina@redhat.com>
Date: Tue, 25 Jul 2023 13:48:10 -0300
Subject: [PATCH] UPSTREAM: <carry>: disable new tests for alpha feature
 NodeLogQuery

---
 openshift-hack/e2e/annotate/rules.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/openshift-hack/e2e/annotate/rules.go b/openshift-hack/e2e/annotate/rules.go
index e77b30c56a5..8a044f7039e 100644
--- a/openshift-hack/e2e/annotate/rules.go
+++ b/openshift-hack/e2e/annotate/rules.go
@@ -148,6 +148,9 @@ var (
 			// https://issues.redhat.com/browse/OCPBUGS-13392
 			`\[sig-network\] NetworkPolicyLegacy \[LinuxOnly\] NetworkPolicy between server and client should enforce policy to allow traffic only from a pod in a different namespace based on PodSelector and NamespaceSelector`,
 			`\[sig-network\] NetworkPolicyLegacy \[LinuxOnly\] NetworkPolicy between server and client should enforce updated policy`,
+
+			// https://issues.redhat.com/browse/OCPBUGS-16760
+			`\[Feature:NodeLogQuery\]`,
 		},
 		// tests that may work, but we don't support them
 		"[Disabled:Unsupported]": {
-- 
2.41.0

