From 5da3a82410f460cf6fe7702623d1ca023849b1ae Mon Sep 17 00:00:00 2001
From: Fabio Bertinatto <fbertina@redhat.com>
Date: Mon, 24 Mar 2025 17:29:35 -0300
Subject: [PATCH] UPSTREAM: <carry>: Add OpenShift tooling, images, configs and
 docs

---
 openshift-hack/e2e/include.go | 1 -
 1 file changed, 1 deletion(-)

diff --git a/openshift-hack/e2e/include.go b/openshift-hack/e2e/include.go
index 48efbca4a3e..e0b481a17f0 100644
--- a/openshift-hack/e2e/include.go
+++ b/openshift-hack/e2e/include.go
@@ -10,7 +10,6 @@ package e2e
 import (
 	// define and freeze constants
 	_ "k8s.io/kubernetes/test/e2e/feature"
-	_ "k8s.io/kubernetes/test/e2e/nodefeature"
 
 	// test sources
 	_ "k8s.io/kubernetes/test/e2e/apimachinery"
-- 
2.49.0

